pkg/session/syssession/pool.go

```diff
@@ -265,6 +281,23 @@ func (p *AdvancedSessionPool) WithSession(fn func(*Session) error) error {
 		}
 	}()
 
+	// Make sure the internal session is registered to the session manager to block GC.
+	const retryInterval = 100 * time.Millisecond
+	if forceBlockGC && !infosync.ContainsInternalSession(se.internal.sctx) {
+		for !infosync.StoreInternalSession(se.internal.sctx) {
+			// In most cases, the session manager is not set, so this step will be skipped.
+			// It is only enabled explicitly in tests through a failpoint.
+			forceBlockGCInTest := !intest.InTest
+			failpoint.Inject("ForceBlockGCInTest", func(val failpoint.Value) {
```

Comment:
The failpoint inject can inject a function logic for test. Maybe it is better than now
```
failpoint.Inject("MockForceBlockGCInTest", func() {
    if (xxxx)
      for (xxx) {
    time.Sleep(retryInterval)
}
})


```